app/api/tutorials/[id]/steps/[stepNumber]/route.ts

```
@@ -0,0 +1,147 @@
+      const lines = fileRaw.split(/\r?\n/);
+      const startIdx = Math.max(0, (desc.from ? desc.from - 1 : 0));
+      const endIdx = Math.min(lines.length, desc.to ? desc.to : lines.length);
+      const content = lines.slice(startIdx, endIdx).join('\n');
+      snippets.push({ ...desc, content });
+    }
+
+    // 1) Parse <CodeFromFiles snippets={[{...}, {...}]}/> blocks
+    // Robust parser that balances braces to extract snippets={[ ... ]}
+    const filesTagRegex = /<CodeFromFiles\s+([^>]*?)\/>/g;
+    let filesTagMatch: RegExpExecArray | null;
+    while ((filesTagMatch = filesTagRegex.exec(parsed.content)) !== null) {
```

Biome (2.1.2)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)